packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -180,6 +182,28 @@ export class TransformerResolver implements TransformerResolverProvider {
             const tableName = this.datasource.ds.dynamoDbConfig?.tableName;
             dataSource = `$util.qr($ctx.stash.put("tableName", "${tableName}"))`;
           }
+
+          const syncConfig = SyncUtils.getSyncConfig(context, this.typeName);
+          if (syncConfig) {
```

Comment:
use datastore check here as well
```suggestion
          if (ctx.isProjectUsingDataStore()) {
            const syncConfig = SyncUtils.getSyncConfig(context, this.typeName) as SyncConfig;
```